src/coreclr/src/tools/Common/JitInterface/CorInfoTypes.cs

```diff
@@ -1029,6 +1029,27 @@ public unsafe struct CORINFO_CALL_INFO
         public bool wrapperDelegateInvoke { get { return _wrapperDelegateInvoke != 0; } set { _wrapperDelegateInvoke = value ? (byte)1 : (byte)0; } }
     }
 
+    public unsafe struct CORINFO_DEVIRTUALIZATION_INFO
+    {
+        //
+        // [In] arguments of resolveVirtualMethod
+        //
+        public CORINFO_METHOD_STRUCT_* virtualMethod;
+        public CORINFO_CLASS_STRUCT_* objClass;
+        public CORINFO_CONTEXT_STRUCT* context;
+
+        //
+        // [Out] results of resolveVirtualMethod.
+        // - devirtualizedMethod is set to MethodDesc of devirt'ed method iff we were able to devirtualize.
+        //      invariant is `resolveVirtualMethod(...) == (devirtualizedMethod != nullptr)`.
+        // - requiresInstMethodTableArg is set to TRUE if the devirtualized method requires a type handle arg.
+        // - exactContext is set to wrapped CORINFO_CLASS_HANDLE of devirt'ed method table.
+        //
+        public CORINFO_METHOD_STRUCT_* devirtualizedMethod;
+        public uint _requiresInstMethodTableArg;
+        public bool requiresInstMethodTableArg { get { return _requiresInstMethodTableArg != 0; } set { _requiresInstMethodTableArg = value ? (byte)1 : (byte)0; } }
```

Comment:
There is some entropy wrt passing bool flags between managed and unmanaged code in this file. So far I see `uint`s and `byte`s used. You probably copied that pattern from `CORINFO_CALL_INFO` above which uses Windows `BOOL` on native side. Other structs defined in `corinfo.h` have fields declared as c++ `bool`.

Reply:
I noticed -- @davidwrighton can you comment on that?

Reply:
Yes, `bool` is `byte` in managed code and `BOOL` is `uint` in managed code. (And to be pedantic, VARIANT_BOOL is ushort in managed code, but just don't use that type.) Due to the various alignment rules in play the code as written will function correctly on our current set of platforms, but please change the field type to byte.

@Rattenkrieg is completely correct. In a perfect world we'd standardize on bool. In fact, I'd like to see us change our c++ headers so that they are compatible with a standard set of C++ headers instead of requiring the PAL headers, but that's a much larger change that I may attempt over the holidays.